aiida/tools/importexport/dbexport/__init__.py

```diff
@@ -223,9 +224,31 @@ def export_tree(
     if not silent:
         print('RETRIEVING LINKED NODES AND STORING LINKS...')
 
-    to_be_exported, links_uuid, graph_traversal_rules = retrieve_linked_nodes(
-        given_calculation_entry_ids, given_data_entry_ids, **kwargs
-    )
+    initial_nodes_ids = given_calculation_entry_ids.union(given_data_entry_ids)
+    traverse_output = get_nodes_export(starting_pks=initial_nodes_ids, get_links=True, **kwargs)
+    to_be_exported = traverse_output['nodes']
+    graph_traversal_rules = traverse_output['rules']
+
+    # I create a utility dictionary for mapping pk to uuid.
```

Comment:
This is done already some other place in this function.
Please try to realign this so we're not doing it several times.

So it's fine to do it here, but then it shouldn't be done again later.

Reply:
But maybe it is fine to do it like this, and then I'll try to patch the export function with a separate PR later.